packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -489,13 +565,178 @@ class SliderThemeData extends Diagnosticable {
+  ///
+  /// [thumbCenter] is the offset of the center of the thumb to the origin of
+  /// the [PaintingContext.canvas]. It can be used as the point that divides
+  /// the track into 2 segments.
+  ///
+  /// [isEnabled] is whether or not the [Slider] is enabled. The enabled state
+  /// is the [Slider.isInteractive] state
+  ///
+  /// [isDiscrete] is whether or not the [Slider] is discrete. The slider is
+  /// discrete when [Slider.divisions] is non-null.
+  void paint(
+    PaintingContext context,
+    Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
+    Animation<double> enableAnimation,
+    TextDirection textDirection,
+    Offset thumbCenter,
+    bool isEnabled,
+    bool isDiscrete,
+  });
+}
+
+/// Base class for slider tick mark shapes.
+///
+/// Create a subclass of this if you would like a custom slider tick mark shape.
+/// This is a simplified version of [SliderComponentShape] with a
+/// [SliderThemeData] passed when getting the preferred size.
+///
+/// See also:
+///
+///  * [RoundSliderTickMarkShape] for a simple example of a tick shape.
+///  * [SliderTrackShape] for the base class for custom a track shape.
+///  * [SliderComponentShape] for the base class for custom a component shape.
+abstract class SliderTickMarkShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderTickMarkShape();
+
+  /// Returns the preferred size of the shape.
+  ///
+  /// It is used to help position the tick marks within the slider.
+  ///
+  /// [sliderTheme] is the theme assigned to the [Slider] that this track shape
+  /// belongs to.
+  ///
+  /// [isEnabled] is whether or not the [Slider] is enabled.
+  Size getPreferredSize({
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+  });
+
+  /// Paints the shape, taking into account the state passed to it.
+  ///
+  /// [context] is the context of the render box of the [Slider]. The origin of
+  /// the canvas in this context is what the [center] is relative to.
+  ///
+  /// [center] is the offset of the where the center of the thumb should
+  /// be painted. This offset is relative to the origin of the [context] canvas.
+  ///
+  /// [parentBox] is the render box within the [Slider] itself. It can be used
+  /// to help determine the preferredRect relative to attributes of the render
+  /// box of the slider itself, such as size.
+  ///
+  /// [sliderTheme] is the theme assigned to the [Slider] that this track shape
+  ///  belongs to.
+  ///
+  /// [enableAnimation] is an animation triggered when the [Slider] is enabled,
+  /// which is defined by [Slider.isInteractive]. It reverses when the slider
+  /// is disabled.
+  ///
+  /// [isEnabled] is whether or not the [Slider] is enabled. The enabled state
+  /// is the [Slider.isInteractive] state
+  ///
+  /// [onActiveTrack] is whether or not this tick mark is on an active part of
```

Comment:
Please don't just reiterate the name of a property as its definition.

Reply:
This param no longer exists, removed

Reply:
Moved a lot of these things to macros too